shared/js/background/events.es6.js

```diff
@@ -266,35 +266,46 @@ startup.ready().then(() => {
     )
 })
 
+// Store the created tab id for when onBeforeNavigate is called so data can be copied across from the source tab
+const createdTargets = new Map()
 browser.webNavigation.onCreatedNavigationTarget.addListener(details => {
-    const currentTab = tabManager.get({ tabId: details.sourceTabId })
-    if (currentTab && currentTab.adClick) {
-        const newTab = tabManager.createOrUpdateTab(details.tabId, { url: details.url })
-        if (currentTab.adClick.shouldPropagateAdClickForNewTab(newTab)) {
-            newTab.adClick = currentTab.adClick.propagate(newTab.id)
-        }
-    }
+    createdTargets.set(details.tabId, details.sourceTabId)
 })
 
 /**
  * Web Navigation
  */
 // keep track of URLs that the browser navigates to.
 //
-// this is currently meant to supplement tabManager.updateTabUrl() above:
+// this is supplemented by tabManager.updateTabUrl() on headersReceived:
 // tabManager.updateTabUrl only fires when a tab has finished loading with a 200,
 // which misses a couple of edge cases like browser special pages
 // and Gmail's weird redirect which returns a 200 via a service worker
-browser.webNavigation.onCommitted.addListener(details => {
+browser.webNavigation.onBeforeNavigate.addListener(details => {
     // ignore navigation on iframes
     if (details.frameId !== 0) return
 
-    const tab = tabManager.get({ tabId: details.tabId })
-
-    if (!tab) return
+    const currentTab = tabManager.get({ tabId: details.tabId })
+    const newTab = tabManager.create({ tabId: details.tabId, url: details.url })
```

Comment:
After `browser.webNavigation.onBeforeNavigate` requests can still happen from the previous document (e.g. things triggered on `onbeforeunload` and things using the `Beacon` API). Those will now probably get attributed to the next page.